pkg/serviceaccount/openidmetadata.go

```diff
@@ -75,7 +78,17 @@ func (p *openidConfigProvider) Enqueue() {
 	}
 }
 func (p *openidConfigProvider) Update() error {
-	pubKeys := p.pubKeyGetter.GetPublicKeys("")
+	pubKeys := []externaljwt.PublicKey{}
+
+	timedCtx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel()
```

Comment:
we don't want this to trigger long-running fetches, we expect this to only return locally cached keys, so pass in a pre-canceled context

Reply:
Still needs to be fixed.

Reply:
Done